### PR TITLE
Feature Implement Form Builder

### PR DESCRIPTION
**Pull request by AbdullahAhmedSoomro**
- Form Builder Implemented.

### CODE REVIEW

File: lib/presentation/screen/intro_screen.dart [hunk 1]
```diff
@@ -7,9 +7,17 @@ import 'package:star_book/presentation/widgets/gradient_scaffold.dart';
 import 'package:star_book/presentation/routes/app_router_name.dart';
 import 'package:star_book/presentation/utils/padding_style.dart';
 
-class IntroScreen extends StatelessWidget {
+class IntroScreen extends StatefulWidget {
```
Comment (hashirshoaeb) [inline lib/presentation/screen/intro_screen.dart:10]: why stateful? are you using setstate?

File: lib/presentation/screen/intro_screen.dart [hunk 2]
```diff
@@ -54,4 +72,16 @@ class IntroScreen extends StatelessWidget {
...
+    final name = RegExp(r'^[a-zA-Z]+$');
+    if (value == null || value.isEmpty) {
+      return 'Please enter your name';
+    } else if (value.length > 15) {
+      return 'name should be less than 15 characters';
+    } else if (!name.hasMatch(value)) {
+      return 'name should only contain alphabetic characters';
+    }
+    return null;
+  }
```
Comment (hashirshoaeb) [inline lib/presentation/screen/intro_screen.dart:86]: more to seperate class as in Qt validator class.

File: lib/presentation/screen/journal_screens/journal_edit_screen.dart [hunk 3]
```diff
@@ -30,23 +38,37 @@ class JournalEditScreen extends StatelessWidget {
...
-                  moodColor: Colors.green,
-                  mood: 'Productive'),
+                date: '05 September 2022',
+                moodColor: Colors.green,
+                mood: 'Productive',
+              ),
               const SizedBox(height: 30),
-              const CustomTextFormField(
+              CustomTextFormField(
+                fieldKey: 'EditTitle',
```
Comment (hashirshoaeb) [inline lib/presentation/screen/journal_screens/journal_edit_screen.dart:47]: create a form model for keys as in Qt

File: lib/presentation/shared/form_validator.dart [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  /// [FormValidator] that requires a field to be non nullable
+  static FormFieldValidator<T> Function<T>({String? errorText}) required =
+      FormBuilderValidators.required;
+
+  /// [FormValidator] that requires the length of the field's value to be
+  /// greater than or equal to the provided minimum length.
+  static FormFieldValidator<T> Function<T>(int minLength, {String? errorText})
+      minLength = FormBuilderValidators.minLength;
+}
```
Comment (hashirshoaeb) [inline lib/presentation/shared/form_validator.dart:17]: nice. move name validator here as well

**Review by hashirshoaeb — CHANGES_REQUESTED** [review]
left some comments.

File: lib/presentation/screen/intro_screen.dart (hunk 1 above)
Comment (AbdullahAhmedSoomro) [reply lib/presentation/screen/intro_screen.dart:10]: I always do this type of mistake 🤦‍♂️

Comment (hashirshoaeb) [reply lib/presentation/screen/intro_screen.dart:10]: hehe don't be so sloppy.

**Review by hashirshoaeb — APPROVED** [review]

**Pull request merged**